infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -252,16 +239,16 @@ public TokenType txnCanCommit(TxResolutionInfo txInfo, /** Input. */
                         log.debug("ABORT[{}] conflict-key[{}](ts={})", txInfo, conflictParam, v);
                         conflictKey.set(conflictParam);
                         response.set(TokenType.TX_ABORT_CONFLICT);
+                        break;
                     }
 
-                    if (v == null && maxConflictWildcard > txSnapshotTimestamp) {
-                        log.warn("ABORT[{}] conflict-key[{}](WILDCARD ts={})", txInfo,
-                                conflictParam,
-                                maxConflictWildcard);
-                        conflictKey.set(conflictParam);
-                        response.set(TokenType.TX_ABORT_CONFLICT);
+                    if (txSnapshotTimestamp < maxConflictWildcard) {
+                        log.debug("ABORT[{}] snapshot-ts[{}] WILDCARD ts=[{}]",
+                                txInfo, txSnapshotTimestamp, maxConflictWildcard);
+                        response.set(TX_ABORT_SEQ_OVERFLOW);
```

Comment:
This does not differentiate between TRIM and cache overflow. I guess these are different error conditions and might have different remedial actions. If a transaction is aborted because of trim, it might be because some transaction is running too slow or some client running that transaction is too slow. If it is a seq overflow the system might have bursts and we do not have enough memory in the server to handle those.